List expression lite plus format trailing changes

mmatera opened the pull request:
In this PR #311 is merged over #308

rocky commented on mathics/builtin/box/inout.py:
```diff
@@ -465,41 +485,41 @@ def apply_list(self, boxes, evaluation):
 
     def to_expression(self):
         """
         returns an evaluable expression.
```
I meant to mention this from the #311. Please let us make the same change were we use the simpler more natural `->Expression` rather than gobbledigook. 

However, where a comment would be most appreciated is in line 522 with something along the lines of why we need to use `Expression()` and not `RowBox()` so that someone (probably me) in the future doesn't get confused again. 

Also, if this is the merge of ListExpression "lite", then ListExpression(*items) ot `to_mathics_list()` neither work?

rocky commented on mathics/builtin/box/inout.py:
```diff
@@ -172,8 +180,20 @@ def render(format, string, in_text=False):
+    def get_head(self) -> Symbol:
+        return SymbolString
+
+    def get_head_name(self) -> "str":
```
Removing the quotes here would be nice since they aren't needed.

rocky commented:
A number of places in test code that number now appear with `<mn>` tags instead of `<mtext>`. 

However I don't see exactly where that is in the code. And it would be nice to mention this in CHANGES as well.

mmatera commented:
@rocky, Since #308 was merged, this branch now does not have sense. In incorporated changes according to the comments in #311.

Regarding the change ``<mn>``<->``<mtext>``, it was just to reproduce the behavior of WMA as faithful as possible. I added a line in CHANGES mentioning the change in the behavior, and added a comment in `test/format/test_format.py` regarding this.
